app/wasm/test/custom_query_test.go

```diff
@@ -43,6 +43,27 @@ func SetupCustomApp(t *testing.T, addr sdk.AccAddress) (*app.OsmosisApp, sdk.Con
 	return osmosis, ctx
 }
 
+func TestQueryFullDenom(t *testing.T) {
+	actor := RandomAccountAddress()
+	osmosis, ctx := SetupCustomApp(t, actor)
+
+	reflect := instantiateReflectContract(t, ctx, osmosis, actor)
+	require.NotEmpty(t, reflect)
+
+	// query full denom
+	query := wasmbindings.OsmosisQuery{
+		FullDenom: &wasmbindings.FullDenom{
+			Contract: actor.String(),
```

Comment:
This test is correct.
But to make more sense, I would use `reflect.String()` here and also in line 63 (expectation).

The contract that will make a mint message would call this